packages/react/src/components/dialog/index.tsx

```diff
@@ -42,18 +42,33 @@ export const Dialog = ({
   container,
   dismissable = true,
   modal = true,
-}: Props) => {
+}: DialogProps) => {
+  /**
+   * Check if the container function return value changed every cycle but only
+   * trigger another render when it actually changes.
+   */
+  const containerFuncReturn = (() => {
+    if (typeof container !== "function") return null
+
+    return container()
+  })()
+
+  const __container = useMemo(() => {
```

Comment:
@jake-slashid I see the comment but I am not sure if I understand the concern here. Are you making sure that we don't accidentally send a new `container` prop value with each render cycle?

Reply:
Yes but I realise now that we could simplify this to:
```tsx
<RadixDialog.Portal
container={
  typeof container === "function"
    ? container()
    : container
}>
```

Because the dom node reference should be stable.

I think at the time for some reason I was concerned about the function reference causing renders.

Reply:
Changed, is that better?

Reply:
Yes, thank you!